### PR TITLE
SDK-1546 evict transactional HTTP client connection pool

### PR DESCRIPTION
**Pull request by kleewho**
fix: Evict connections after reestablishing connection

Eviction of OkHttp connection pool after reestablishing connection (affects transactional calls)

### CODE REVIEW

**Comment by kleewho** [comment]
It was originally published as #223 and I merged it into #221 thinking that we're going to merge them both soon. We haven't so to save time of reviewers I'm publishing this once again

**Review by seba-aln — DISMISSED** [review]
LGTM

File: src/main/java/com/pubnub/api/managers/RetrofitManager.java [hunk 1]
```diff
@@ -104,6 +107,20 @@ public RetrofitManager(PubNub pubNubInstance) {
                             this.pubnub.getConfiguration().getConnectTimeout()
                     ).retryOnConnectionFailure(false)
             );
+
+            this.pubnub.addListener(new SubscribeCallback.BaseSubscribeCallback() {
+                @Override
```
Comment (650elx) [inline src/main/java/com/pubnub/api/managers/RetrofitManager.java:112]: I understand that this solution needs to be hacky, so IMO we need to have a detailed description for that change in merge commit.

File: src/main/java/com/pubnub/api/managers/RetrofitManager.java [hunk 2]
```diff
@@ -23,11 +26,11 @@
...
 import okhttp3.OkHttpClient;
 import okhttp3.logging.HttpLoggingInterceptor;
+import org.jetbrains.annotations.*;
 import retrofit2.Retrofit;
 
-import java.util.Collections;
-import java.util.concurrent.ExecutorService;
-import java.util.concurrent.TimeUnit;
+import java.util.*;
+import java.util.concurrent.*;
```
Comment (650elx) [inline src/main/java/com/pubnub/api/managers/RetrofitManager.java:33]: Do we have to use wildcard imports?

File: .github/workflows/run_acceptance_tests.yml [hunk 3]
```diff
@@ -20,21 +20,31 @@ jobs:
         uses: ./client-engineering-deployment-tools/actions/mock-server
```
Comment (650elx) [inline .github/workflows/run_acceptance_tests.yml:20]: Smaller PRs with tight scope next time please ;)

**Review by 650elx — DISMISSED** [review]

File: .github/workflows/run_acceptance_tests.yml (hunk 3 above)
Comment (kleewho) [reply .github/workflows/run_acceptance_tests.yml:20]: Yes. Sorry for that. I did trim a lot if you compare it with #221 :)

File: src/main/java/com/pubnub/api/managers/RetrofitManager.java (hunk 1 above)
Comment (kleewho) [reply src/main/java/com/pubnub/api/managers/RetrofitManager.java:112]: Yes. I'm going to add comment in the code as well

File: src/main/java/com/pubnub/api/managers/RetrofitManager.java (hunk 2 above)
Comment (kleewho) [reply src/main/java/com/pubnub/api/managers/RetrofitManager.java:33]: We don't

**Review by 650elx — APPROVED** [review]

**Review by bartk — DISMISSED** [review]

**Comment by kleewho** [comment]
@client-engineering-bot release v5.2.3

**Comment by parfeon** [comment]
@client-engineering-bot release as v5.2.3

**Review by bartk — APPROVED** [review]

**Review by kazydek — APPROVED** [review]

**Pull request merged**